k8s/k8s.go

```diff
@@ -0,0 +1,291 @@
+			events = append(events, osv.Event{LastAffected: av.Introduced})
+		}
+		ranges = append(ranges, osv.Range{
+			Events: events,
+		})
+		affected = append(affected, osv.Affected{Ranges: ranges, Package: osv.Package{Name: p, Ecosystem: "kubernetes"}, Severities: []osv.Severity{{Type: cvss.Type, Score: cvss.Vector}}})
+	}
+	return affected
+}
+
+func getComponentName(k8sComponent string, mitreComponent string) string {
+	if len(k8sComponent) == 0 {
+		k8sComponent = mitreComponent
+	}
+	if strings.ToLower(mitreComponent) != "kubernetes" {
+		k8sComponent = mitreComponent
+	}
+	return strings.ToLower(fmt.Sprintf("%s/%s", upstreamOrgByName(k8sComponent), upstreamRepoByName(k8sComponent)))
+}
+
+func validateCvesData(cves []*osv.OSV) error {
+	var result error
+	for _, cve := range cves {
+		if len(cve.ID) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nid is mssing on cve #%s", cve.ID))
+		}
+		if len(cve.Published) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nCreatedAt is mssing on cve #%s", cve.ID))
+		}
+		if len(cve.Summary) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nSummary is mssing on cve #%s", cve.ID))
+		}
+		for _, af := range cve.Affected {
+			if len(strings.TrimPrefix(af.Package.Name, upstreamOrgByName(af.Package.Name))) == 0 {
+				result = errors.Join(result, fmt.Errorf("\nComponent is mssing on cve #%s", cve.ID))
+			}
+		}
+		if len(cve.Details) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nDescription is mssing on cve #%s", cve.ID))
+		}
+		if len(cve.Affected) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nAffected Version is missing on cve #%s", cve.ID))
+		}
+		if len(cve.Affected) > 0 {
+			for _, v := range cve.Affected {
+				for _, s := range v.Severities {
+					if len(s.Type) == 0 {
+						result = errors.Join(result, fmt.Errorf("\nVector is mssing on cve #%s", cve.ID))
+					}
+				}
+				for _, r := range v.Ranges {
+					for i := 1; i < len(r.Events); i++ {
+						if len(r.Events[i-1].Introduced) == 0 {
+							result = errors.Join(result, fmt.Errorf("\nAffectedVersion Introduced is missing from cve #%s", cve.ID))
+						}
+						if len(r.Events[i].Fixed) == 0 && len(r.Events[i].LastAffected) == 0 {
+							result = errors.Join(result, fmt.Errorf("\nAffectedVersion Fixed and LastAffected are missing from cve #%s", cve.ID))
+						}
+					}
+				}
+			}
+		}
+		if len(cve.References) == 0 {
+			result = errors.Join(result, fmt.Errorf("\nUrls is mssing on cve #%s", cve.ID))
+		}
+	}
+	return result
+}
+
+func cveMissingImportantData(vulnerability *Cve) bool {
+	return len(vulnerability.versions) == 0 ||
+		len(vulnerability.Package) == 0 ||
+		len(vulnerability.CvssV3.Vector) == 0
+}
+
+// getExitingCvesToModifiedMap get the existing cves from vuln-list-k8s repo and map it to cve id and last updated
+func getExitingCvesToModifiedMap() (map[string]string, error) {
+	response, err := http.Get(vulnListRepoTarBall)
```

Comment:
It is already [stored](https://github.com/aquasecurity/vuln-list-update/blob/166257a854438736fe0240ecb0ac809bfa108419/.github/workflows/k8s.yml#L24-L29) locally. Do we need to download it again?

Reply:
I would prefer to not couple between code logic to github action, wdyt ?

Reply:
Agree. I had the same feeling, but it already depends on the git dir. If vuln-list-k8s doesn't exist, it will fail. I'm suggesting the reuse of that dir.
https://github.com/aquasecurity/vuln-list-update/blob/2927f2b673cd17083870806ed120cffd80cada41/k8s/k8s.go#L83

`vuln-list-update` assumes the vuln-list dir exists. The dir path can be passed from outside.
https://github.com/aquasecurity/vuln-list-update/blob/2927f2b673cd17083870806ed120cffd80cada41/main.go#L43

What if writing Makefile like trivy-db? The benefit is the source code can be much simpler.
https://github.com/aquasecurity/trivy-db/blob/b7e0a5886daa539ee175c666ecf051c2da94ef05/Makefile#L91-L100

Reply:
I get your point, you convinced me , I think it will be simpler , I'll implement it with existing as suggested initially

Reply:
fixed.